src/GraphNodeManagerViewExtension/ViewModels/GridNodeViewModel.cs

```diff
@@ -388,21 +380,10 @@ private bool IsNodeEmptyList(MirrorData mirrorData)
         /// <returns></returns>
         private bool IsNodeOutputFunction(MirrorData mirrorData)
         {
-            if (mirrorData == null || mirrorData.Class == null) return false;
+            if (mirrorData.IsNull || mirrorData == null || mirrorData.Class == null) return false;
 
-            try
-            {
-                if (string.Equals(mirrorData.Class.Name, "Function")) return true;
-                return false;
-            }
-            catch (NullReferenceException)
-            {
-                return false;
-            }
-            catch (Exception)
-            {
-                return false;
-            }
+            if (string.Equals(mirrorData.Class.Name, "Function")) return true;
```

Comment:
I would keep the try catch, or also check name for null/empty

Reply:
Thanks for pointing that out! I added a ``IsNullOrEmpty`` check for the mirrorData.Class.Name before using it as per your suggestion, @mjkkirschner.